tests/devices/test_default_qubit_autograd.py

```diff
@@ -541,3 +543,120 @@ def test_partial_subsystem(self, mocker):
+            multiplication with scalars and addition.
+            """
+
+            def __init__(self, val):
+                self.val = val
+
+            def __mul__(self, other):
+                new = SpecialObject(self.val)
+                new *= other
+                return new
+
+            def __imul__(self, other):
+                self.val *= other
+                return self
+
+            def __rmul__(self, other):
+                return self * other
+
+            def __iadd__(self, other):
+                self.val += other.val if isinstance(other, self.__class__) else other
+                return self
+
+            def __add__(self, other):
+                new = SpecialObject(self.val)
+                new += other.val if isinstance(other, self.__class__) else other
+                return new
+
+            def __radd__(self, other):
+                return self + other
+
+        class SpecialObservable(Observable):
+            """SpecialObservable"""
+
+            num_wires = AnyWires
+            num_params = 0
+            par_domain = None
+
+            def diagonalizing_gates(self):
+                """Diagonalizing gates"""
+                return []
+
+        class DeviceSupportingSpecialObservable(DefaultQubit):
+            name = "Device supporting SpecialObservable"
+            short_name = "default.qubit.specialobservable"
+            observables = DefaultQubit.observables.union({"SpecialObservable"})
+
+            @classmethod
+            def capabilities(cls):
+                capabilities = super().capabilities().copy()
+                capabilities.update(
+                    provides_jacobian=True,
+                )
+                return capabilities
+
+            def expval(self, observable, **kwargs):
+                if self.analytic and isinstance(observable, SpecialObservable):
+                    val = super().expval(qml.PauliZ(wires=0), **kwargs)
+                    return SpecialObject(val)
+
+                return super().expval(observable, **kwargs)
+
+            def jacobian(self, tape):
+                # we actually let pennylane do the work of computing the
+                # jacobian for us but return it as a device jacobian
+                gradient_tapes, fn = qml.gradients.param_shift(tape)
+                tape_jacobian = fn(qml.execute(gradient_tapes, self, None))
+                return tape_jacobian
+
+        dev = DeviceSupportingSpecialObservable(wires=1, shots=None)
+
+        # force diff_method='parameter-shift' because otherwise
+        # PennyLane swaps out dev for default.qubit.autograd
+        @qml.qnode(dev, diff_method="parameter-shift")
+        def qnode(x):
+            qml.RY(x, wires=0)
+            return qml.expval(SpecialObservable(wires=0))
+
+        @qml.qnode(dev, diff_method="parameter-shift")
+        def reference_qnode(x):
+            qml.RY(x, wires=0)
+            return qml.expval(qml.PauliZ(wires=0))

```

Comment:
Could we substitute this part with providing the analytic expression for the jacobian? 

I think we should have the value in our test suite somewhere and it would:
1. Not rely on the logic of PennyLane to check something;
2. Not require extra QNode evaluations.

Reply:
hmm, I think it's clearer as it is in showing that it is "equivalent" to the sample pl circuit.